### ISSUE TITLE
Are 1D Convolutions supported?

### ISSUE THREAD

**Issue by georgesterpu** [intel/neural-compressor#1135]
Hi,

When attempting to enable the Weight Pruning callback with a "channelx1" pattern for a Pytorch model using mainly Conv1D operations, the neural-compressor library throws the following error:
```
RuntimeError: shape '[1, 128, 128, 1]' is invalid for input of size 81920
```
This is triggered when the [`get_reduced_masks_from_data` method](https://github.com/intel/neural-compressor/blob/master/neural_compressor/compression/pruner/patterns/nxm.py#L117) is called for a Conv1D module with a weight shaped `[128, 128, 5]`.
Looking at the code in `_reshape_orig_to_2dims()`, I can see that it expects the weight to have 4 dimensions, i.e. it assumes this is a Conv2D weight, otherwise it does not reshape it to `[cout, kx*ky*cin]`, hence the incorrect shape.

Could you please confirm if the weights of Conv1d operations are supported by the weight pruner?
Does the main workaround at this stage consist in replacing all the 1D convolutions in my code with equivalent 2D convolutions?

Thank you!

**Comment by chensuyue** [comment]
@WeiweiZhang1  please take a look at this question.

**Comment by wenhuach21** [comment]
@n1ck-guo  please also take transformer.conv1d into account, which is different from torch.conv1d

**Comment by n1ck-guo** [comment]
torch.conv1d and transformer.conv1d are already supported in the latest version Intel Neural Compressor.

**Comment by georgesterpu** [comment]
Thanks for the update, @n1ck-guo 
Do you have an estimate date for the upcoming neural-compressor binary release on PyPI including the conv1d commit? I see that the last version available there is v2.2.1, released on Jul 21.

**Comment by n1ck-guo** [comment]
@georgesterpu v2.3 will release on September 15.
